apply/cloud_applier.go

```diff
@@ -35,13 +35,42 @@ func NewAliCloudProvider(cluster *v1.Cluster) Interface {
 	return &CloudApplier{d}
 }
 
+func (c *CloudApplier) ScaleDownNodes(cluster *v1.Cluster) (isScaleDown bool, err error) {
+	logger.Info("desired master %s, current master %s, desired nodes %s, current nodes %s", c.ClusterDesired.Spec.Masters.Count,
+		cluster.Spec.Masters.Count,
+		c.ClusterDesired.Spec.Nodes.Count,
+		cluster.Spec.Nodes.Count)
+	if c.ClusterDesired.Spec.Masters.Count >= cluster.Spec.Masters.Count &&
```

Comment:
the invoking chain is too long, would it be better to declare the values at the beginning? just a suggestion

Reply:
I think it's OKay.